### ISSUE TITLE
[minifier] Accuracy minification

### ISSUE THREAD

**Issue by anijain2305** [pytorch/pytorch#93715]
- [ ] For integer tensors, currently we have torch.zeros for rand_strided (https://github.com/pytorch/torchdynamo/blob/main/torchdynamo/testing.py#L254). This is ok for Exceptions, but not really useful for accuracy. We need to get the maximum value from the original tensor and then replace zeros with randint(max_value)

* `git revert 35590bb9ae5781d93e483540e9492e1409347800`
* `TORCHDYNAMO_REPRO_AFTER="dynamo" TORCHDYNAMO_REPRO_LEVEL=4 benchmarks/huggingface.py -d cuda --inductor --training --float32 --accuracy  --only=BertForQuestionAnswering` shows that accuracy divergence was observed
* But `python /tmp/minifier_anijain/minifier_launcher.py` does not minify
* Changing torch.zeros to torch.randint(10, ..) as mentioned above would fix it. So, the goal is to get the max_value plumbed through.

- [ ] We currently do fwd_output.sum().backward() - This could hide indexing issues, because the loss is same for all the tensors. We should provide torch.rand_like() input gradients. More context here - https://github.com/pytorch/torchdynamo/pull/1414#issuecomment-1262801213

cc @ezyang @msaroufim @wconstab @bdhirsh @zou3519 @soumith @ngimel @williamwen42 @yanboliang
